### PR TITLE
Homepage and scoreboard

### PR DESCRIPTION
**Pull request by matous-volf**
- Moved the default /public to /public/scoreboard.
- Created a temporary homepage for /public.
- Created a blog controller and a temporary blog post.

### CODE REVIEW

File: webapp/src/Controller/BlogController.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace App\Controller;
+
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+#[Route(path: '/public/blog')]
+class BlogController extends BaseController
+{
+    #[Route(path: '/{id}', name: 'public_blogpost')]
```
Comment (zapotocnylubos) [inline webapp/src/Controller/BlogController.php:11]: pridal bych route na "/" pro seznam vsech prispevku

potom route s povinnym "/{id}", kdyz by nekdo chtel primo referencovat jednotlivy prispevek

File: webapp/src/Controller/PublicController.php [hunk 2]
```diff
@@ -39,6 +39,11 @@ public function __construct(
     ) {}
 
     #[Route(path: '', name: 'public_index')]
+    public function homepageAction(): Response {
+        return $this->render('public/homepage.html.twig');
+    }
+
+    #[Route(path: '/scoreboard', name: 'public_scoreboard')]
```
Comment (zapotocnylubos) [inline webapp/src/Controller/PublicController.php:46]: jsou doopravdy podchycene vsechny prejmenovani? 
public_index -> public_scoreboard

File: webapp/templates/public/homepage.html.twig [hunk 3]
```diff
@@ -0,0 +1,60 @@
+{% extends "public/base.html.twig" %}
+
+{% block title %}Korespondenční seminář{% endblock %}
+{% block content %}
```
Comment (zapotocnylubos) [inline webapp/templates/public/homepage.html.twig:4]: zkus tomu dat alespon basic styly, dat to do containeru nebo vyuzit nejake komponenty z bootstrapu, ktery DJ pouziva

File: webapp/src/Controller/BlogController.php (hunk 1 above)
Comment (zapotocnylubos) [reply webapp/src/Controller/BlogController.php:11]: => teda sablonu pro seznam novinek (ten blog)
=> sablonu na deail prispevku

File: webapp/templates/public/menu.html.twig [hunk 4]
```diff
@@ -11,8 +11,8 @@
     <div class="collapse navbar-collapse" id="menuDefault">
         <ul class="navbar-nav me-auto">
             <li class="nav-item">
-                <a class="nav-link{% if current_route in ['public_index', 'public_team'] %} active{% endif %}"
-                    href="{{ path('public_index') }}"><i class="fas fa-list-ol"></i> Scoreboard</a>
+                <a class="nav-link{% if current_route in ['public_scoreboard', 'public_team'] %} active{% endif %}"
+                    href="{{ path('public_scoreboard') }}"><i class="fas fa-list-ol"></i> Scoreboard</a>
             </li>
             <li class="nav-item">
```
Comment (zapotocnylubos) [inline webapp/templates/public/menu.html.twig:17]: pridat do navigace link na blog

File: webapp/templates/public/homepage.html.twig [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    <h2>Aktuality</h2>
+    <h3>Curabitur ligula sapien</h3>
+    <p>1. 9. 2023</p>
+    <p>
+        Maecenas lorem. Mauris metus. Aliquam erat volutpat. Nam sed tellus id magna elementum tincidunt. Pellentesque
+        pretium lectus id turpis. Integer imperdiet lectus quis justo. Vestibulum erat nulla, ullamcorper nec, rutrum
+        non, nonummy ac, erat. Vivamus porttitor turpis ac leo. In enim a arcu imperdiet malesuada. Integer pellentesque
+        quam vel velit. Phasellus enim erat, vestibulum vel, aliquam a, posuere eu, velit.
+    </p>
+    <a href="/public/blog/1" class="btn btn-primary mb-3">Číst dál</a>
```
Comment (zapotocnylubos) [inline webapp/templates/public/homepage.html.twig:25]: na homepage nebudou novinky, ale spis general info

File: webapp/templates/public/homepage.html.twig (hunk 3 above)
Comment (matous-volf) [reply webapp/templates/public/homepage.html.twig:4]: container-fluid tam jiz je z /templates/base.
zkusim si pohrat s marginem a treba pouzit cards.

File: webapp/src/Controller/PublicController.php (hunk 2 above)
Comment (matous-volf) [reply webapp/src/Controller/PublicController.php:46]: mely by byt. nechal jsem jen cache a [RootController](https://github.com/delta-cs/seminar-domjudge/blob/1e6001b1a76cf6647fd9b0dc1ecc639b5531f6d4/webapp/src/Controller/RootController.php#L39).

Comment (zapotocnylubos) [reply webapp/src/Controller/PublicController.php:46]: jakou cache myslis?

Comment (matous-volf) [reply webapp/src/Controller/PublicController.php:46]: /webapp/var/cache (gitignornute).
a [link na DJ logu v navigaci](https://github.com/delta-cs/seminar-domjudge/blob/1e6001b1a76cf6647fd9b0dc1ecc639b5531f6d4/webapp/templates/public/menu.html.twig#L3) jsem taky nechal.

File: webapp/templates/public/homepage.html.twig (hunk 3 above)
Comment (matous-volf) [reply webapp/templates/public/homepage.html.twig:4]: aha, myslel jsem, ze do containeru se dalsi container nedava, ale jo. :)
takze pridam.

File: webapp/src/Controller/PublicController.php (hunk 2 above)
Comment (zapotocnylubos) [reply webapp/src/Controller/PublicController.php:46]: jo, oboje spravne

File: webapp/templates/public/menu.html.twig (hunk 4 above)
Comment (zapotocnylubos) [reply webapp/templates/public/menu.html.twig:17]: nechano na dalsi MR

File: webapp/src/Controller/BlogController.php (hunk 1 above)
Comment (zapotocnylubos) [reply webapp/src/Controller/BlogController.php]: blog bude v jinem MR

**Comment by zapotocnylubos** [comment]
@matous-volf prihodis desktop a mobile screenshoty ten homepage prosim?

**Comment by matous-volf** [comment]
image

**Comment by matous-volf** [comment]
image

**Comment by matous-volf** [comment]
image

**Review by zapotocnylubos — APPROVED** [review]

**Pull request merged**